src/models/base.ts

```diff
@@ -1,32 +1,31 @@
-import type { Constructor } from "./utils";
+import type { Constructor, InferModelData, InferModelMetadata } from "./utils";
 import type { DetailedAsyncAPI } from "../types";
 
 export interface ModelMetadata {
   asyncapi: DetailedAsyncAPI;
   pointer: string;
-  [key: string]: any;
 }
 
-export abstract class BaseModel<J extends Record<string, any> = Record<string, any>, M extends Record<string, any> = {}> {
+export abstract class BaseModel<J extends any = any, M extends Record<string, any> = {}> {
   constructor(
     protected readonly _json: J,
-    protected readonly _meta: ModelMetadata & M = {} as any,
+    protected readonly _meta: ModelMetadata & M = {} as ModelMetadata & M,
   ) {}
 
-  json<T extends Record<string, any> = J>(): T;
+  json<T = J>(): T;
   json<K extends keyof J>(key: K): J[K];
   json(key?: keyof J) {
-    if (key === undefined) return this._json;
+    if (key === undefined || typeof this._json !== 'object') return this._json;
```

Comment:
I don't get why we need this change. what type could be other than object? If the user provides a key, it is expected to only return a value.

Reply:
The value can also be different from the object - we have one case in our spec, schema can have a boolean, true or false value.

Reply:
But what if the user provides a value for `key`? You should still assume the user wants the value for such a key, no matters the type of `this._json`.

Reply:
Right, I will fix it.

Reply:
Done, new function body:

```ts
    if (key === undefined) return this._json;
    if (this._json === null || this._json === undefined) return this._json;
    return this._json[key];
```